kythe/java/com/google/devtools/kythe/extractors/java/standalone/Javac9Wrapper.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2014 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.devtools.kythe.extractors.java.standalone;
+
+import com.google.common.base.Optional;
+import com.google.devtools.kythe.extractors.java.JavaCompilationUnitExtractor;
+import com.google.devtools.kythe.extractors.shared.CompilationDescription;
+import com.sun.tools.javac.file.JavacFileManager;
+import com.sun.tools.javac.main.Arguments;
+import com.sun.tools.javac.main.Option;
+import com.sun.tools.javac.util.Context;
+import com.sun.tools.javac.util.Options;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardLocation;
+
+/** A class that wraps javac to extract compilation information and write it to an index file. */
+public class Javac9Wrapper extends AbstractJavacWrapper {
+  @Override
+  protected CompilationDescription processCompilation(
+      String[] arguments, JavaCompilationUnitExtractor javaCompilationUnitExtractor)
+      throws Exception {
+
+    // Use javac's argument parser to get the list of source files
+    Context context = new Context();
+
+    JavacFileManager fileManager = new JavacFileManager(context, true, null);
+    Arguments args = Arguments.instance(context);
+    args.init("kythe_javac", arguments);
+    fileManager.handleOptions(args.getDeferredFileManagerOptions());
+    Options options = Options.instance(context);
+
+    List<String> sources = new ArrayList<>();
```

Comment:
If it doesn't need to be mutable, maybe

List<String> sources = args.getFileObjects().stream().map(fo -> fo.getName()).collect(toImmutableList());

Or not, I'm fine this way too.